lib/Page.js

```diff
@@ -775,20 +775,20 @@ class Page extends EventEmitter {
    * @param {!Array<string>} values
    */
   async select(selector, ...values) {
-    await this.$eval(selector, (element, values) => {
+    return await this.$eval(selector, (element, values) => {
       if (element.nodeName.toLowerCase() !== 'select')
         throw new Error('Element is not a <select> element.');
-
       const options = Array.from(element.options);
-
+      const filteredValues = values.filter(value => options.some(option => option.value === value));
       if (element.multiple) {
         for (const option of options)
-          option.selected = values.includes(option.value);
+          option.selected = filteredValues.includes(option.value);
       } else {
-        element.value = values.shift();
+        element.value = filteredValues[0];
```

Comment:
... and this line can be just 
```js
// explicitly assign `undefined` to state that passing in empty array resets selection.
element.value = values.length ? values[0] : undefined;
```

Reply:
I thought it would be cool if the function selected the first value in values that also exists in options. I can do it in one loop like this: `element.value = undefined;
      for (var i = values.length - 1; i>=0; i--){
        for (var option of options){
          if (option.value === values[i]){
            option.selected = true;
            break;
          }
        }
      }`

Reply:
I thought it would be nice if the function would select the first value in values that also exists in options. I can do it in one loop like this if youd prefer.
```
element.value = undefined;
      for (var i = values.length - 1; i>=0; i--){
        for (var option of options){
          if (option.value === values[i]){
            option.selected = true;
            break;
          }
        }
      }
```

Reply:
> I thought it would be nice if the function would select the first value in values that also exists in options. I can do it in one loop like this if youd prefer.

Can you please clarify why do you want this? I don't think this is of much importance, so I'd rather have code as simple as possible.

Reply:
I thought that was the only way it would make sense to pass multiple values to this function when the found select element only takes one value. You could create an array of values, order them by priority, and the function would select the value that exists with highest priority.

Otherwise, I think passing multiple values into this function when the select element only takes one value is a logical flaw. When someone just sets the select element to be the first value, if there is no such value in options it deselects everything. So you would have asked the function to select multiple options, but it actually deselects every option.

It does increase code complexity, so I understand if you disagree.

Reply:
@AlexChung1995 Wouldn't the following achieve the same effect you're proposing?

```javascript
async select(selector, ...values) {
  return await this.$eval(selector, (element, values) => {
    if (element.nodeName.toLowerCase() !== 'select')
      throw new Error('Element is not a <select> element.');

    const options = Array.from(element.options);

    for (const option of options) {
      option.selected = values.includes(option.value);

      if ((option.selected === true) && (element.multiple === false)) {
        break;
      }
    }

    element.dispatchEvent(new Event('input', { 'bubbles': true }));
    element.dispatchEvent(new Event('change', { 'bubbles': true }));

    return options.filter(option => option.selected).map(option => option.value);
  }, values);
}
```

Reply:
Yeah, I considered that option but thought you guys wouldn't like it because of the if statement in the loop. Out of the 3 implementations which would you prefer? The pick the first value original, my version of the new heuristic, or your version of the new heuristic? Let me know and Ill update the code and commit.

Reply:
@AlexChung1995 thanks, I understand now your intentions here.

> Otherwise, I think passing multiple values into this function when the select element only takes one value is a logical flaw.

I agree. We shouldn't encourage this, so I'd avoid giving any special treatment to the multiple values passed in for a non-multiple select.

Given this, I like the second suggestion with dropped if-statement:

```js
async select(selector, ...values) {
  return await this.$eval(selector, (element, values) => {
    if (element.nodeName.toLowerCase() !== 'select')
      throw new Error('Element is not a <select> element.');

    const options = Array.from(element.options);
    for (const option of options) 
      option.selected = values.includes(option.value);

    element.dispatchEvent(new Event('input', { 'bubbles': true }));
    element.dispatchEvent(new Event('change', { 'bubbles': true }));
    return options.filter(option => option.selected).map(option => option.value);
  }, values);
}
```

It's dummy, which is nice.

Reply:
Ok, Ill do another commit. Since this is your solution @alixaxel I assume you're ok with it? 
Thanks @alixaxel @aslushnikov for your help.